ledger/complete/wal/checkpointer.go

```diff
@@ -187,6 +187,15 @@ type SyncOnCloseFile struct {
 	*bufio.Writer
 }
 
+func (s *SyncOnCloseFile) Sync() error {
+	err := s.Flush()
+	if err != nil {
+		return fmt.Errorf("cannot flush buffer: %w", err)
+	}
+	return s.file.Sync()
+
+}
+
```

Comment:
I feel we need something like this, I noticed most of the time our flaky test might be fixed by calling the sync.